decisions/0003-tagstore-removal.rst

```diff
@@ -0,0 +1,31 @@
+Status
+======
+
+Accepted
+
+
+Context
+=======
+
+During the initial development of Blockstore for the LabXchange project, a tagging service called tagstore was
+developed to allow tagging XBlock content stored in Blockstore. However rapidly evolving product needs made it
+very complicated to have a performant generic tagging system which could be used by multiple systems. Therefore
+tagging functionality was added to the LabXchange backend service instead.
```

Comment:
The reason was more specific than that - it was to store the Taxonomies as versioned data in bundles:

> Currently [in Tagstore], taxonomies of tags are stored in MySQL and use "materialized paths" to represent hierarchy where present. Presumably, when XBlock/course content in Blockstore is moved, duplicated, deleted, etc., Tagstore will listen for signals from Blockstore and then update its database accordingly. One problem I have yet to solve is how to include tags in the bundle when exporting courses etc. for import into other Open edX instances (we want to reference the taxonomy that the tags come from, and import it as well but not if it's already present). I also realize that the current implementation makes it difficult for draft/publish support with tags - tags are always applied immediately since Tagstore isn't aware of draft content vs. published content.
>
> I realized today that a potential solution to most of these issues is **to store the tag data in files in a Bundle instead of in MySQL**. I had initially rejected that option because it doesn't support hierarchical querying of the tag data (searching for "animal" should return things tagged "lion"), but I've since realized that we'll likely want to index the tag data in ElasticSearch anyways, and as long as we index the materialized paths in ElasticSearch, it doesn't matter how they're stored in the original tag/taxonomy data source.
>
> ...
> So we're talking about a dedicated Bundle type that is a Taxonomy, that's edited, published, and linked to by other Bundles? I like it.

Reply:
@bradenmacdonald Updated. Does this work?

Reply:
@symbolist Looks good, thanks!